x-pack/plugins/observability/public/hooks/slo/mix_kql_with_tags.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchState } from '../../pages/slos/hooks/use_url_search_state';
+
+export function mixKqlWithTags(kqlQuery: string, tags: SearchState['tags']) {
+  if (!tags) {
+    return kqlQuery;
+  }
+  const tagsKqlIncluded = tags.included?.join(' or ') || '';
+  const excludedTagsKql = tags.excluded?.join(' or ') || '';
+
+  let tagsQuery = '';
+  if (tagsKqlIncluded && excludedTagsKql) {
+    tagsQuery = `slo.tags: (${excludedTagsKql}) and not slo.tags: (${tagsKqlIncluded})`;
+  }
+  if (!excludedTagsKql && tagsKqlIncluded) {
+    tagsQuery = `slo.tags: (${tagsKqlIncluded})`;
+  }
+  if (!tagsKqlIncluded && excludedTagsKql) {
+    tagsQuery = `not slo.tags: (${excludedTagsKql})`;
+  }
+
+  if (!kqlQuery) {
+    return tagsQuery;
+  }
+
+  if (tagsQuery) {
```

Comment:
@kdelemme These lines fix the issue. Before it was:
```
return `${kqlQuery} and ${tagsQuery}`;
```
Before the fix, in case user has cleared out the tags and let's say for example search query was `something` I was getting `something and` and this was giving me no results in the UI. 

Not sure though if my fix is the proper fix.